src/extension/wps/wps-core/src/main/java/org/geoserver/wps/gs/GetCoveragesValue.java

```diff
@@ -0,0 +1,230 @@
+import org.geoserver.catalog.CoverageInfo;
+import org.geoserver.data.util.CoverageUtils;
+import org.geoserver.wps.WPSException;
+import org.geotools.api.coverage.grid.Format;
+import org.geotools.api.coverage.grid.GridCoverageReader;
+import org.geotools.api.geometry.Position;
+import org.geotools.api.parameter.ParameterValueGroup;
+import org.geotools.api.referencing.FactoryException;
+import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
+import org.geotools.api.referencing.datum.PixelInCell;
+import org.geotools.api.referencing.operation.MathTransform;
+import org.geotools.api.referencing.operation.TransformException;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.GridEnvelope2D;
+import org.geotools.coverage.grid.GridGeometry2D;
+import org.geotools.coverage.grid.io.AbstractGridFormat;
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.gce.imagemosaic.ImageMosaicFormat;
+import org.geotools.geometry.GeneralBounds;
+import org.geotools.geometry.Position2D;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.image.ImageWorker;
+import org.geotools.process.factory.DescribeParameter;
+import org.geotools.process.factory.DescribeProcess;
+import org.geotools.process.factory.DescribeResult;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Coordinate;
+
+/** A process that returns values from a coverage at a given location */
+@DescribeProcess(title = "GetCoveragesValue", description = "Returns values from a coverage at a given location")
+public class GetCoveragesValue implements GeoServerProcess {
+    private final Catalog catalog;
+
+    public GetCoveragesValue(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    @DescribeResult(name = "result", description = "Output value or values at location", type = ValuesAtPoint.class)
+    public ValuesAtPoint execute(
+            @DescribeParameter(
+                            name = "name",
+                            description =
+                                    "Comma delimited names of rasters, optionally fully qualified (workspace:name), will check for the first one that intersects the point and returns a value")
+                    String name,
+            @DescribeParameter(name = "x", description = "x coordinate of the location to sample") double x,
+            @DescribeParameter(name = "y", description = "y coordinate of the location to sample") double y,
+            @DescribeParameter(
+                            name = "crs",
+                            description =
+                                    "Coordinate Reference System of the x and y coordinates, defaults to first coverage CRS if not set",
+                            min = 0,
+                            max = 1)
+                    String crs)
+            throws IOException {
+        CoverageInfo ci = null;
+        CoordinateReferenceSystem coordinateReferenceSystem = null;
```

Comment:
Could you declare variables close to their first point of usage? And then see if any of the initial "validate and acquire actions" could be factored out in their own methods.
I'm commenting on this one occurrence but there are more in the code that follows.

Reply:
done